gamestonk_terminal/stocks/options/yfinance_view.py

```diff
@@ -662,3 +666,238 @@ def risk_neutral_vals(
     else:
         print(new_df.to_string(index=False))
     print("")
+
+
+def plot_expected_prices(
+    und_vals: List[List[float]], p: float, ticker: str, expiration: str
+) -> None:
+    """Plot expected prices of the underlying asset at expiration
+
+    Parameters
+    ----------
+    und_vals : List[List[float]]
+        The expected underlying values at the expiration date
+    p : float
+        The probability of the stock price moving upward each round
+    ticker : str
+        The ticker of the option's underlying asset
+    expiration : str
+        The expiration for the option
+    """
+    up_moves = list(range(len(und_vals[-1])))
+    up_moves.reverse()
+    probs = [binom.pmf(r, len(up_moves), p) for r in up_moves]
+    fig, ax = plt.subplots(figsize=plot_autoscale(), dpi=cfp.PLOT_DPI)
+    ax.set_title(f"Probabilities for ending prices of {ticker} on {expiration}")
+    ax.xaxis.set_major_formatter("${x:1.2f}")
+    ax.yaxis.set_major_formatter(mtick.PercentFormatter())
+    plt.plot(und_vals[-1], probs)
+    fig.tight_layout()
+    if gtff.USE_ION:
+        plt.ion()
+    plt.show()
+
+
+def export_binomial_calcs(
+    up: float,
+    prob_up: float,
+    discount: float,
+    und_vals: List[List[float]],
+    opt_vals: List[List[float]],
+    days: int,
+    ticker: str,
+) -> None:
+    """Create an excel spreadsheet with binomial tables for underlying asset value and option value
+
+    Parameters
+    ----------
+    up : float
+        The stock's increase on an upward move
+    prob_up : float
+        The probability of an upward move
+    discount : float
+        The daily discount rate
+    und_vals : List[List[float]]
+        The underlying asset values at each step
+    opt_vals : List[List[float]]
+        The values for the option at each step
+    days : int
+        The number of days until the option expires
+    ticker : str
+        The ticker for the company
+    """
+    letters = [
+        "A",
+        "B",
+        "C",
+        "D",
+        "E",
+        "F",
+        "G",
+        "H",
+        "I",
+        "J",
+        "K",
+        "L",
+        "M",
+    ]
+    letters += [
+        "N",
+        "O",
+        "P",
+        "Q",
+        "R",
+        "S",
+        "T",
+        "U",
+        "V",
+        "W",
+        "X",
+        "Y",
+        "Z",
+    ]
+    opts = (
+        [f"{x}" for x in letters]
+        + [f"{x}{y}" for x in letters for y in letters]
+        + [f"{x}{y}{z}" for x in letters for y in letters for z in letters]
+    )
+    wb = Workbook()
+    ws = wb.active
+
+    ws["A1"] = "U"
+    ws["B1"] = up
+    ws["A2"] = "D"
+    ws["B2"] = 1 / up
+    ws["D1"] = "q"
+    ws["E1"] = prob_up
+    ws["D2"] = "1-q"
+    ws["E2"] = 1 - prob_up
+    ws["D3"] = "Discount"
+    ws["E3"] = discount
```

Comment:
More descript names here as well.  When I open up an xls and see q (which is usually denoted as 1-probability of success), I have no idea what it means